docs/e2e-testing.md

```diff
@@ -35,9 +35,7 @@ func genericTestSuite(t *testing.T) {
 
     // Create the operatorsource.
     err = helpers.CreateRuntimeObject(f, ctx, helpers.CreateOperatorSource(namespace))
-    if err != nil {
-        t.Errorf("Could not create operator source: %v", err)
-    }
+    require.NoError(t, err, "Could not create OperatorSource")
```

Comment:
Any reason to not print the error using `fmt.Sprintf("Could not create OperatorSource: %v", err)`? Comment applies to all places.

Reply:
@aravindhp I believe the test framework prints out the error in a standard format if the assertion or require fail. in fact, I am still of the opinion that we don't need any custom error messages at all, but I we decided to keep them.

Reply:
Please post a snippet of the output.

Reply:
@aravindhp I changed an `assert.True` to `assert.False` without any custom output message and this is the log from the logger: 
```
FAIL: TestMarketplace (165.64s)
    --- FAIL: TestMarketplace/operator-source-test-group (98.38s)
    	client.go:57: resource type OperatorSource with namespace/name (openshift-marketplace/test-operators) created
        --- FAIL: TestMarketplace/operator-source-test-group/opsrc-creation-test-suite (25.29s)
            --- FAIL: TestMarketplace/operator-source-test-group/opsrc-creation-test-suite/operator-source-generates-expected-objects (25.29s)
            	opsrccreationtests.go:43: 
            			Error Trace:	opsrccreationtests.go:43
            			Error:      	Should be false
            			Test:       	TestMarketplace/operator-source-test-group/opsrc-creation-test-suite/operator-source-generates-expected-objects
```

Reply:
I forced an error before a `require`: 
```
err = k8s_errors.NewAlreadyExists(schema.GroupResource{}, "This is a test error")
require.NoError(t, err)
```
and this is the output: 
```--- FAIL: TestMarketplace (145.67s)
    --- FAIL: TestMarketplace/operator-source-test-group (83.33s)
        client.go:57: resource type OperatorSource with namespace/name (openshift-marketplace/test-operators) created
        --- FAIL: TestMarketplace/operator-source-test-group/opsrc-creation-test-suite (20.26s)
            --- FAIL: TestMarketplace/operator-source-test-group/opsrc-creation-test-suite/operator-source-generates-expected-objects (20.26s)
                require.go:794:
                        Error Trace:    opsrccreationtests.go:35
                        Error:          Received unexpected error:
                                         "This is test error" already exists
                        Test:           TestMarketplace/operator-source-test-group/opsrc-creation```

Reply:
Given these output, I vote for not having custom error messages as they can get too verbose and the output from these error point exactly to the `assert`/`require` statement that's failing which is efficient enough.

Reply:
I disagree. This will force us to go look at the source code to figure out what is wrong. If there is a properly worded error messages, we can figure out what is wrong at a glance. This will be important especially in the CI pipeline and for folks who are not familiar with the operator. But if others agree with you I will take back my objection.

Reply:
@aravindhp updated PR with `fmt.Sprintf` at all places.

Reply:
Quick question - I reran the test locally:

```
func qTest(t *testing.T) {
	err := fmt.Errorf("this is my error")
	require.NoError(t, err, fmt.Sprintf("error: %v", err))

	/* OUTPUT
	Error Trace:	main_test.go:72
	Error:      	Received unexpected error:
					this is my error
	Test:       	TestMarketplace/test
	Messages:   	error: this is my error
	*/
}
```

It looks like the message is printed twice in the current format. Is this intended?

Reply:
Interesting. One of them is from the test framework, and the other is the `custom` message. According to @aravindhp's point, the error from the test framework can be used to hunt down the error in the code, while the `custom` error message will help out in CI pipeline. 
@awgreene does the output look different if you use `require.NoError(t, err, "this is my error")` instead?

Reply:
I believe the example provided ealier, https://github.com/operator-framework/operator-marketplace/pull/177#discussion_r284726972, failed to add a message which is why this slipped through.

Reply:
>....@awgreene does the output look different if you use require.NoError(t, err, "this is my error") instead?

```
func qTest(t *testing.T) {
	err := fmt.Errorf("this is my error")
	//require.NoError(t, err, fmt.Sprintf("error: %v", err))
	require.NoError(t, err, "this is my error")
	/* OUTPUT
	Error Trace:	main_test.go:73
	Error:      	Received unexpected error:
					this is my error
	Test:       	TestMarketplace/test
	Messages:   	this is my error

	*/
}
```

Reply:
If the output provides the error, I am fine leaving the error out from the message. In that case `fmt.Sprintf()` should only be used in cases where we trying to show some other variable for additional informational. But I am still a +1 on custom error messages.

Reply:
@awgreene in [#177(comment)](https://github.com/operator-framework/operator-marketplace/pull/177#discussion_r284726972) the message has been left out intentionally to showcase what the test framework prints `without` a `custom` error message.

Reply:
> @awgreene in #177(comment) the message has been left out intentionally to showcase what the test framework prints without a custom error message.

Ah thank you

Reply:
> If the output provides the error, I am fine leaving the error out from the message. In that case fmt.Sprintf() should only be used in cases where we trying to show some other variable for additional informational. But I am still a +1 on custom error messages.


+1

Reply:
@aravindhp @awgreene updated PR to have custom error messages, while not repeating the same error message in the log. Let me know your thoughts.